src/controller/python/chip/setup_payload/setup_payload.py

```diff
@@ -113,3 +118,51 @@ def __InitNativeFunctions(self, chipLib):
         setter.Set("pychip_SetupPayload_PrintOnboardingCodes",
                    c_int32,
                    [c_uint32, c_uint16, c_uint16, c_uint16, uint8_t, uint8_t, uint8_t])
+
+    # Getters from parsed contents.
+    # Prefer using the methods below to access setup payload information once parse.
+
+    @property
+    def vendor_id(self) -> int:
+        return int(self.attributes.get("VendorID", "0"))
+
+    @property
+    def product_id(self) -> int:
+        return int(self.attributes.get("ProductID", "0"))
+
+    @property
+    def setup_passcode(self) -> int:
+        if "SetUpPINCode" not in self.attributes:
+            raise KeyError("Missing setup passcode in setup payload: parsing likely not yet done")
+
+        return int(self.attributes["SetUpPINCode"])
+
+    @property
+    def long_discriminator(self) -> Optional[int]:
+        if "Long discriminator" not in self.attributes:
+            return None
+
+        return int(self.attributes["Long discriminator"])
+
+    @property
+    def short_discriminator(self) -> Optional[int]:
+        if "Short discriminator" not in self.attributes:
+            return None
+
+        return int(self.attributes["Short discriminator"])
```

Comment:
Does something already init the short discriminator from an available long discriminator?  Or is the expectation that in the "we have a long discriminator" case this will return None and the consumer should derive the short discriminator from the long one themselves?

Reply:
The latter: what I found is that it's either the short or long discriminator present, since it's using whichever one is present in the *actual* setup payload.

Already when doing discovery, it's important state to the API whether you have short or long discriminator. You can't just do "long == short << 8". Rather you must ignore the lower 8 bits and only match top 4 when you have short. I did not try to fix that.